docs/guides/using-the-network-shim.md

```diff
@@ -0,0 +1,193 @@
+-   `stub`: in this mode the Network Shim also intercepts all outgoing requests to the specified hosts, but it will prevent the request from reaching the destination server and instead it will produce a response from a previously recorded fixture.
+-   `live`: the Network Shim does not intercept any requests when this mode is enabled.
+
+By first recording fixtures during a _capture run_ and then using these fixtures during a _stub run_ it effectively becomes possible to execute a full cypress test suite without a live backend. This is especially relevant for running a cypress test suite on CI.
+
+A capture or stub run is executed for a specific DHIS2 api version and individual fixtures for each version are produced/used during a run. This paves the way for doing cross-version testing for _feature-toggling_ apps.
+
+## General workflow
+
+As mentioned, he Network Shim is a tool that enables a workflow where a cypress e2e test suite is executed on CI.
+
+The general steps required to setup a workflow like that are as follows:
+
+1. Ensure a live backend is up and running
+1. Ensure a fully passing cypress test-suite is in place
+1. Ensure the GitHub workflow includes a job that runs the cypress test suite on CI in stub mode
+1. Execute a capture run, which will produce fixture files
+1. Commit the generated fixture files and push to the remote
+1. The test suite will now execute on CI and if a change breaks a test it will be impossible to merge
+
+To setup cross-version testing essentially means taking some of the steps above multiple times:
+
+-   Perform multiple capture runs against different versions of the live backend, so network fixtures are created for all these versions
+-   Define multiple jobs in the GitHub workflow that execute a stub run against these different versions
+
+## Configuration
+
+A workflow as described above consists of a few moving parts, which are described in more details below. **However, running the `d2-utils-cypress install` command should create all required files**.
+
+### Cypress
+
+#### `./cypress.env.json`
+
+The network shim expects a few environment variables to be present. These can be set using the `--env` flag when interacting with the cypress CLI, but base values should be provided as well the `./cypress.env.json` configuration file:
+
+```json
+{
+    "dhis2Username": "YOUR_USERNAME_HERE",
+    "dhis2Password": "YOUR PASSWORD HERE",
+    "dhis2BaseUrl": "http://localhost:8080",
+    "networkMode": "live",
+    "dhis2ApiVersion": "37"
+}
+```
+
+#### `./cypress.json`
+
+Interactive runs are not formally supported by the Network Shim, because `cypress open` (interactive mode) and `cypress run` have different behaviour regarding the plugin events. However, cypress is attempting to smooth things over and by setting `experimentalInteractiveRunEvents` to true it should theoretically be possible to at at least use stub mode in interactive mode, for example:
```

Comment:
This is a little unclear to me. If I understand what you're saying correctly, maybe you can phrase it as:

> Stubbing partial runs is not supported

I think I wouldn't even go in depth in documenting something that we don't officially support. Just mentioning that we don't support it seems good enough to me. Otherwise maybe it'll only get confusing.

Reply:
Yeah... I struggled with this a little bit... I like having the `experimentalInteractiveRunEvents ` toggled to `true` in the config because it makes it more likely that things will work correctly when using `cypress open` instead of `cypress run`. So I'd like to keep that field in.... But since it's there, staring people in the face, I felt like I should also provide an explanation. And that's how I ended up writing this. Would you recommend just removing the explanation?

Reply:
I think the changes you've made have resolved this for me